lib/puppet/provider/ibm_pkg/imcl.rb

```diff
@@ -69,17 +69,35 @@ def imcl_command_path
     @imcl_command_path
   end
 
+
+  # searches for installed.xml in potential appDataLocation dirs
+  #
+  # @return [String] installed.xml path if it is found
+  def self.find_installed_xml(user)
+    require 'find'
+
+    root_path = "/var/ibm/"
```

Comment:
So, two things.
1) You'll need to do a File.exist? before you do the find, otherwise it will throw an error if the path doesn't exist.
2) You can simplify this with:
```
user_path = if user == 'root' ? '/var/ibm' : "/home/#{user}/var/ibm"

if File.exist? user_path
  Find.find(user_path) { |path| installed_xml_path = path if path.match(/InstallationManager\/installed.xml$/) }
end

installed_xml_path if File.file?(installed_xml_path)